tests/Progress/ProgressHashesTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace BristolSU\Support\Tests\Progress;
+
+use BristolSU\Support\Progress\ProgressHashes;
+use BristolSU\Support\Tests\TestCase;
+use Illuminate\Foundation\Testing\WithFaker;
+
+class ProgressHashesTest extends TestCase
+{
+    use WithFaker;
+
+    protected $Model;
+    protected $Table;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->Model = (new ProgressHashes);
+        $this->Table = $this->Model->getTable();
+    }
+
+    public function getModelData()
+    {
+        return [
+            'item_key' => $this->faker('en_UK')->slug,
+            'hash' => $this->faker('en_UK')->uuid
+        ];
+    }
+
+    /** @test */
+    public function a_model_can_be_saved_and_retrieved_by_the_hash(){
+        $data = $this->getModelData();
+
+        $this->Model->create($data);
+
+        $this->assertDatabaseHas($this->Table, $data);
+
+        $dbValue = $this->Model->getByHash($data['hash'])->first(['item_key', 'hash'])->toArray();
```

Comment:
This links to my comment on getByHash, I didn't realise this would work! Something to do with a model having the query object stored on it? 

I think I'd still prefer a scope, mainly because I've never seen this method in documentation before. If it's referenced in the docs then fine to keep.

Having written this,  I just ran the test suite and got the exception `Call to a member function first() on null` as I'd expect, so I think change to a scope!